node/pod.go

```diff
@@ -185,6 +186,9 @@ func (pc *PodController) updatePodStatus(ctx context.Context, podFromKubernetes
 	kPod.Lock()
 	podFromProvider := kPod.lastPodStatusReceivedFromProvider.DeepCopy()
 	kPod.Unlock()
+	if reflect.DeepEqual(podFromKubernetes.Status, podFromProvider.Status) {
```

Comment:
Any reason not to use cmp here?

Reply:
Thanks for review. 

Good idea, updated